jupyter-js-widgets/src/widget_string.ts

```diff
@@ -385,3 +385,121 @@ class TextView extends LabeledDOMWidgetView {
 
     textbox: HTMLInputElement;
 }
+
+export
+class PasswordModel extends StringModel {
+    defaults() {
+        return _.extend(super.defaults(), {
+            _view_name: 'PasswordView',
+            _model_name: 'PasswordModel'
+        });
+    }
+}
+
+export
+class PasswordView extends LabeledDOMWidgetView {
```

Comment:
Again, here how about just inheriting from `TextView`, and doing something like:

```
render() {
    super.render();
    this.textbox.setAttribute('type', 'password');
}
```
and let everything else be inherited?